server/media/media_test.go

```diff
@@ -0,0 +1,86 @@
+package media
+
+import (
+	"testing"
+)
+
+func TestMatchCORSOrigin(t *testing.T) {
+	cases := []struct {
+		allowed  []string
+		origin   string
+		expected string
+	}{
+		{
+			allowed:  []string{"https://example.com"},
+			origin:   "https://example.com",
+			expected: "https://example.com",
+		},
+		{
+			allowed:  []string{"https://example2.com", "https://example.com"},
+			origin:   "https://example.com",
+			expected: "https://example.com",
+		},
+		{
+			allowed:  []string{"*"},
+			origin:   "https://example.com",
+			expected: "https://example.com",
+		},
+		{
+			allowed:  []string{""},
+			origin:   "https://example.com",
+			expected: "",
+		},
+		{
+			allowed:  []string{},
+			origin:   "https://example.com",
+			expected: "",
+		},
+		{
+			allowed:  []string{"https://example.com"},
+			origin:   "https://sub.example.com",
+			expected: "",
+		},
+		{
+			allowed:  []string{"https://*.example.com"},
+			origin:   "https://sub.example.com",
+			expected: "https://sub.example.com",
+		},
+		{
+			allowed:  []string{"https://*.example.com"},
+			origin:   "https://pre.sub.example.com",
+			expected: "",
+		},
+		{
+			allowed:  []string{"https://*.example.com", "https://*.sub.example.com"},
+			origin:   "https://pre.sub.example.com",
+			expected: "https://pre.sub.example.com",
+		},
+		{
+			allowed:  []string{"https://*.*.example.com"},
+			origin:   "https://pre.sub.example.com",
+			expected: "https://pre.sub.example.com",
+		},
+		{
+			allowed:  []string{"https://*.sub.example.com"},
+			origin:   "https://pre.asd.example.com",
+			expected: "",
+		},
+		{
+			allowed:  []string{"https://pre.*.example.com"},
+			origin:   "https://pre.sub.example.com",
+			expected: "",
+		},
+		{
+			allowed:  []string{"https://*.*.*.example.com"},
+			origin:   "https://www.pre.sub.example.com",
+			expected: "https://www.pre.sub.example.com",
+		},
+	}
+
```

Comment:
I would also add tests for `http` vs `https`, presence of port number in the Origin and Allowed, blank Origin.